### DISCUSSION TITLE
[Feature Request] CSV Results Filter w/ Auto-generated Headers

### DISCUSSION THREAD

**Discussion by briend** [grafana/grafana-infinity-datasource#401] [Ideas]
**Is your feature request related to a problem? Please describe.**
The Results Filter is great, especially since Grafana [does not support variables in transformations.](https://github.com/grafana/grafana/issues/25469) However, the Results Filter box does not appear unless I manually add all of the column headers, even though auto-generation of the headers from the CSV worked fine.

**Describe the solution you'd like**
Present the Results Filter option w/ auto-generated headers

![image](https://user-images.githubusercontent.com/6015639/119385862-a038ac00-bc7b-11eb-8545-05a0fdeebabf.png)

**Comment by yesoreyeram** [comment]
Just wondering how does this makes sense for the auto headers? How the plugin know the field type unless specified.. it can assume the type. But that may be error prone. Amy thoughts? 

I have fix in mind.. but not sure if that is going to work..

**Comment by briend** [comment]
I didn't think too much about the field Types, admittedly.  Maybe it would work to have an option (checkbox?) to combine the auto headers (assuming String) with overrides of one or more manually defined column headers with a Type.

**Comment by pavlozt** [comment]
@yesoreyeram It is assumed that the user specifies one value for Time and declares all others as numeric.

Traditionally, prometheus and grafana use a mass approach:
a few clicks allow you to make a visualization and it continues to work when new time series or objects appears.
However, in infinity plugging, this approach unexpectedly did not work for me.
For example, to work with CSV as a time series, you need to separately assign a column with time and specify the type of each column.
Otherwise, the data is ignored.

How to setup this correctly? And if I can't, maybe  we need to add this feature in the code?
